components/fonts/font_context.rs

```diff
@@ -357,15 +347,104 @@ impl FontContext {
     }
 }
 
-#[derive(Clone)]
 pub(crate) struct WebFontDownloadState {
-    webview_id: WebViewId,
-    pub(crate) css_font_face_descriptors: Arc<CSSFontFaceDescriptors>,
+    webview_id: Option<WebViewId>,
+    css_font_face_descriptors: CSSFontFaceDescriptors,
     remaining_sources: Vec<Source>,
-    finished_callback: WebFontLoadFinishedCallback,
     core_resource_thread: CoreResourceThread,
-    local_fonts: Arc<HashMap<Atom, Option<FontTemplateRef>>>,
-    pub(crate) stylesheet: DocumentStyleSheet,
+    local_fonts: HashMap<Atom, Option<FontTemplateRef>>,
```

Comment:
Could this still be an `Arc` or `Rc` to avoid copying the data?

Reply:
This change is actually avoiding the copy entirely - before this change `WebFontDownloadState` was `Clone` and we were cloning it for each instance of the download, but with this change `WebFontDownloadState` is a move type, so it is created once and ownership is transferred between the `RemoteWebFontDownloader` instances.